java-checks-test-sources/src/main/java/checks/spring/SpringConfigurationWithAutowiredFieldsCheck.java

```diff
@@ -0,0 +1,133 @@
+package checks.spring;
+
+import javax.inject.Inject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+class SpringConfigurationWithAutowiredFieldsCheck {
+  class Bar { }
+
+  class Foo {
+    private final Bar bar;
+    public Foo(Bar bar) { this.bar = bar; }
+  }
+
+  @Configuration
+  class A {
+
+    @Autowired private Bar singleUsage; // Noncompliant [[sc=28;ec=39]] {{Inject this field value directly into "method", the only method that uses it.}}
+    @Inject private Bar jsr330; // Noncompliant [[sc=25;ec=31]] {{Inject this field value directly into "jsr330", the only method that uses it.}}
+    @Autowired private Bar multipleUsage;
+    @Autowired private Bar notUsedInBeanMethod;
+    @Autowired private Bar notUsed;
+    private Bar notAutowired;
+    @Autowired(required=false) private Bar withInitializer = new Bar();
```

Comment:
I'd suggest adding a case `@Autowired private Bar withInitializer = new Bar();` (maybe to non-compiling examples if doesn't compile) and decide what we want to do in such case.